framework/derive/src/parse/attributes/util.rs

```diff
@@ -12,6 +15,66 @@ pub(super) fn is_attribute_with_no_args(attr: &syn::Attribute, name: &str) -> bo
     false
 }
 
+pub(super) fn get_attribute_with_one_type_arg(
+    attr: &syn::Attribute,
+    name: &str,
+) -> Option<EsdtAttribute> {
+    let attr_path = &attr.path;
+    if let Some(first_seg) = attr_path.segments.first() {
+        if first_seg.ident == name {
+            let tokens = attr.tokens.clone();
+            let group = match syn::parse2::<Group>(tokens) {
+                Ok(group) => group.stream(),
+                _ => panic!("Expected a group as attribute argument"),
+            };
+
+            let mut iter = group.into_iter();
+
+            let first_literal = match iter.next() {
+                Some(TokenTree::Literal(literal)) => {
+                    let literal_string = literal.to_string();
+                    literal_string.trim_matches('\"').to_string()
+                },
+                _ => panic!("Expected a literal as the first token in the attribute argument"),
+            };
+
+            let symbol: &'static str = Box::leak(Box::new(first_literal));
```

Comment:
Memory leak!
No need for static str in derive. Use String! The static str is a trick we employ when generating code (might remove it there too...). There's no need to have it here.